Source/Public/Add-VSTeamWorkItemType.ps1

```diff
@@ -0,0 +1,46 @@
+function Add-VSTeamWorkItemType {
+   [cmdletbinding(SupportsShouldProcess=$true)]
+   param(
+      [vsteam_lib.ProcessTemplateValidateAttribute()]
+      [ArgumentCompleter([vsteam_lib.ProcessTemplateCompleter])]
+      $ProcessTemplate = $env:TEAM_PROCESS,
+
+      [parameter(Mandatory = $true)]
+      [Alias('Name')]
+      [string] $WorkItemType,
+
+      [string] $Description,
+
+      [ArgumentCompleter([vsteam_lib.ColorCompleter])]
+      [vsteam_lib.ColorTransformToHexAttribute()]
+      $Color = '000000',
+
+      [vsteam_lib.IconTransformAttribute()]
+      [ArgumentCompleter([vsteam_lib.IconCompleter])]
+      [string]$Icon = 'icon_asterisk'
+   )
+
+   $url =  _getProcessTemplateUrl $ProcessTemplate
+   if ($url) {$url += "/workitemtypes?api-version=" + (_getApiVersion Processes)}
+   else      {Write-Warning "Could not convert '$ProcessTemplate' into a process template"; return}
+
+   $body  = @{
+      name  = $WorkItemType
+      color = $Color
+      icon  = $Icon
+   }
+   if ($Description)  {$body['description'] =$Description }
+
+   if ($PSCmdlet.ShouldProcess($ProcessTemplate, "Add workitem '$WorkItemType' to process template"))  {
+      $resp = _callapi -Url $url -method  Post  -ContentType "application/json" -body (ConvertTo-Json $body) -erroraction stop
```

Comment:
have you tried just handing over the body as an object? In the unit tests, you could then just assert by checking the properties of the psobject.

Reply:
Also, you don't need to add content-type. By default, _callApi is using the UTF-8 content type for JSON, which should work better.

Reply:
I hadn't , I just automatically convert to JSON, and I hadn't seen inside the content type was set inside _callApi. Although I'm fairly sure I've copied that form from other functions, what you're describing would be better.

Reply:
If I pass the body as a pscustomObject  -verbose shows the JSON looks like this  and I get an error
```
VERBOSE: Body @{description=Describes a divergence between required and actual behavior, and tracks the work done to correct the defect and verify the correction.; isDisabled=False; color=ff0000; icon=icon_insect}
````
if I do convertto-json it looks like this 
```
VERBOSE: Body {
  "description": "Describes a divergence between required and actual behavior, and tracks the work done to correct the defect and verify the correction.",
  "isDisabled": false,
  "color": "ff0000",
  "icon": "icon_insect"
}
```
Which has " marks and : instead of = 


But I have taken the `content-type.` out - I also made a change so set works with multiple names,